datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -847,24 +747,33 @@ mod tests {
         let expr_b = binary_expr(lit(1), Operator::Multiply, col("c2"));
         let expected = col("c2");
 
-        assert_eq!(simplify(&expr_a), expected);
-        assert_eq!(simplify(&expr_b), expected);
+        assert_eq!(simplify(expr_a), expected);
+        assert_eq!(simplify(expr_b), expected);
     }
 
     #[test]
     fn test_simplify_divide_by_one() {
         let expr = binary_expr(col("c2"), Operator::Divide, lit(1));
         let expected = col("c2");
 
-        assert_eq!(simplify(&expr), expected);
+        assert_eq!(simplify(expr), expected);
     }
 
     #[test]
     fn test_simplify_divide_by_same() {
         let expr = binary_expr(col("c2"), Operator::Divide, col("c2"));
+        // if c2 is null, c2 / c2 = null, so can't simplify
```

Comment:
new test coverage showing one can't simplify `A / A` unless you know `A` isn't null